analytics/analytics.go

```diff
@@ -184,7 +187,9 @@ func (dc *dataCollection) sendData() {
 
 	res, err := http.DefaultClient.Do(req)
 	if err != nil {
-		res.Body.Close()
+		if res != nil {
+			res.Body.Close()
```

Comment:
You don't need to close body inside an error ...
`On error, any Response can be ignored. A non-nil Response with a non-nil error only occurs when CheckRedirect fails, and even then the returned Response.Body is already closed.`

Reply:
General flow is

```
client := http.DefaultClient
resp, err := client.Do(req)
if err != nil {
    return nil, err
}
defer resp.Body.Close()
```

Reply:
OK, GOT IT

Reply:
@taiyangc By the way, should I close it outside?

Reply:
Yes, outside
Because only when `err == nil`, on error return you don't need to close

Reply:
@SiyingPeng Please refer to example code above, the `defer resp.Body.Close()` is after the error check, and there is no need for `res != nil`

Reply:
@taiyangc done